src/main/webapp/assets/js/youtubePlayerController.js

```diff
@@ -19,75 +19,183 @@ function onYouTubeIframeAPIReady() {
-    */
+    // event.target.loadVideoById("XwxLwG2_Sxk");
+    actionPlayerServlet("START_PLAYER");
 }
 
+// state: -1 = unstarted, 0 = ended, 1 = playing video, 2 = paused,3 = buffering, 5 = video cued
 function onPlayerStateChange(event){
     console.log(getPlayerState());
-    // Check if playing video ended
-    if(getPlayerState() === 0){ 
-        console.log("ENDED");
-        currentVideoId++;
-        // Check if there are more videos in the queue
-        if(currentVideoId < videoIds.length){
-            player.loadVideoById(videoIds[currentVideoId]);
+}
+
+function getPartyId(){
+    // Get party id 5910974510923776
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    return urlParams.get('id');;
+}
+
+// Function that manages communication with sync server
+function syncManager(partyId){
+    // Obtain playlist container to put songs
+    const playlistContainer = document.getElementById('playlist-container');
+    const url = '/musicPlayer?party-id=' + partyId;
+    
+    // Repeat evewrything inside every second
+    window.setInterval(async function () {
+        // Call Servlet
+        const response = await fetch(url);
+        console.log(response);
+
+        switch (response.status) {
+            // All good
+            case 200: 
+                // Get Party playlist state (currentSongPlayInfo, currentPlaylist;)
+                const partyPlaylistState = await response.json();
+
+                // Update playlist
+                getPlaylist(playlistContainer, partyPlaylistState.currentPlaylist);
+
+                // Extract YoutubeSongPlayInfo( YoutubeSong song  / L songStartGmtTimeMs  / Bool stopped)
+                const currentYTSongPlayInfo = partyPlaylistState.currentSongPlayInfo;
+
+                // Extract Youtube song (S songName /S videoId /L songDuration)
+                const currentYTSong = currentYTSongPlayInfo.song;
+                songSync(currentYTSongPlayInfo);
+                
+                document.getElementById("errorMessage").innerHTML = "";
+
+                break;
+            case 400:
+                // Error message - If no room-id is given or a non-numeric room-id is given, return a 400 response code
+                document.getElementById("errorMessage").innerHTML = "Wrong party";
+                break;
+            case 204:
+                // Error message  no songs
+                document.getElementById("errorMessage").innerHTML = "No songs in queue";
+                break;
+            default:
+                // Error
+                document.getElementById("errorMessage").innerHTML = "Lost communication with server, maybe its the aliens";
+                break;
+        }
+    }, 1000);
+}
+
+function songSync(currentSongPlayInfo){
+    const videoId = currentSongPlayInfo.song.videoId;
+    const startTime = currentSongPlayInfo.songStartGmtTimeMs;
+    const currentTime = Date.now();
```

Comment:
Get isStopped also, see if the song should be playing or not.

if(localplayer.ispaused) {
don't play the video;
} if (playing && status.isPaused) {
do stop the video;
}

Reply:
Just so that you can pause the video if you don't want it playing